salt/pillar/__init__.py

```diff
@@ -768,17 +841,35 @@ def _external_pillar_data(self, pillar, val, key):
         Builds actual pillar data structure and updates the ``pillar`` variable
         '''
         ext = None
+        args = inspect.getargspec(self.ext_pillars[key]).args
```

Comment:
@cachedout does six or some other code handle the fact that `inspect.getargspec` is deprecated in python 3?

Reply:
Guh. Good call. I totally forgot about that. IIRC, we've had to gate that behind a PY3-specific check and branch into the proper call.

Reply:
@alexbleotu We need to check if we are running Py3 and if so, switch to [inspect.getfullargspec()](https://docs.python.org/3.6/library/inspect.html#inspect.getfullargspec)

Reply:
We have a convenience function for this: `salt.utils.args.get_function_argspec()`

Reply:
Excellend. Will do

Reply:
Done